### PR TITLE
Add IDX 

### PR DESCRIPTION
**Pull request by domesticmouse**
PTAL @rodydavis

## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md

### CODE REVIEW

**Comment by domesticmouse** [comment]
FWIW, the red CI status has nothing to do with this PR. It will be fixed once #2254 lands and is integrated

File: .idx/dev.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# see: https://developers.google.com/idx/guides/customize-idx-env
+{ pkgs, ... }: {
+  # Which nixpkgs channel to use.
+  channel = "stable-23.11"; # or "unstable"
+
+  # Use https://search.nixos.org/packages to find packages
+  packages = [
+    pkgs.jdk17
+    pkgs.unzip
+    pkgs.flutter
```
Comment (rodydavis) [inline .idx/dev.nix:11]: Not needed as flutter is preinstalled!

File: .idx/dev.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  idx = {
+    # Search for the extensions you want on https://open-vsx.org/ and use "publisher.id"
+    extensions = [
+      "Dart-Code.dart-code"
+      "Dart-Code.flutter"
+    ];
+    workspace = {
+      # Runs when a workspace is first created
+      onCreate = {
+        updateStable = "flutter channel stable && flutter upgrade && flutter pub get";
```
Comment (rodydavis) [inline .idx/dev.nix:25]: Should already be on stable

File: .idx/dev.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      };
+      # Runs when the workspace is (re)started
+      onStart = {
```
Comment (rodydavis) [inline .idx/dev.nix:28]: Only needs to be onCreate

**Review by rodydavis — CHANGES_REQUESTED** [review]

File: .idx/dev.nix (hunk 2 above)
Comment (domesticmouse) [reply .idx/dev.nix:25]: I'm being explicit, with the intent that I can change the branch to `beta` and `main` while doing the maintenance work required to get ready for a new release

**Comment by domesticmouse** [comment]
PTAL @rodydavis

File: .idx/dev.nix (hunk 2 above)
Comment (rodydavis) [reply .idx/dev.nix:24]: You can rerun the onCreate hooks with IDX command palette!

**Review by rodydavis — APPROVED** [review]
LGTM

**Pull request merged**